api/server.js

```diff
@@ -268,6 +274,53 @@ async function checkColor(color_scheme) {
 	return (rows.length >= 1);
 }
 
+/**
+ * Gets all of the books that match the given search query. Each book returned
+ * will include its title, author(s) and isbn number.
+ * 
+ * @param {Object} info Search parameters provided by user
+ * @returns {Object[]} List of books that satify the search query
+ */
+async function getMatchingBooks(info) {
+	let title = info.title;
+	let author = info.author;
+	let genre = info.genre;
+
+	let params = [];
+	let query = `   DROP TEMPORARY TABLE IF EXISTS Results; CREATE TEMPORARY TABLE Results
+					SELECT Books.title AS title, Authors.name AS author_name, Books.ISBN AS isbn
+					FROM Books
+					INNER JOIN Book_Authors
+						ON Books.ISBN = Book_Authors.ISBN
+					INNER JOIN Authors
+						ON Book_Authors.id_author = Authors.id
+					INNER JOIN Book_Genre
+						ON Books.ISBN = Book_Genre.ISBN
+					INNER JOIN Genre
+						ON Book_Genre.id_genre = Genre.id
+					WHERE 1=1
+				`;
+	if (title) {
+		query += " AND Books.title = ?";
+		params.push(title);
+	} 
+	if (genre) {
+		query += " AND Genre.name IN ?";
```

Comment:
My tests fail because of an error in our SQL syntax when I provide a genre search param. We should take a look at this if statement